docker-images/syntax-highlighter/crates/syntax-analysis/src/highlighting/tree_sitter.rs

```diff
@@ -431,6 +432,32 @@ mod test {
         .expect("dump document")
     }
 
+    fn get_language_for_test(filepath: &std::path::Path, contents: &str) -> TreeSitterLanguageName {
+        let language_from_syntect = crate::highlighting::test::SYNTAX_SET
+            .with(|syntax_set| {
+                FileInfo::new(filepath.to_string_lossy().as_ref(), contents, None)
+                    .determine_language(syntax_set)
+            })
+            .unwrap();
+
+        // If we can't determine language from Syntect, determine from path just for the test
+        // This is only needed for test, since when running in production, we
+        // will always have the language passed in
+        if language_from_syntect.raw.is_empty()
+            || language_from_syntect.raw.to_lowercase() == "plain text"
+        {
+            let extension = filepath.extension().unwrap_or_default();
+            if !extension.is_empty() {
+                if let Some(parser_id) = ParserId::from_file_extension(extension.to_str().unwrap())
+                {
+                    return TreeSitterLanguageName::new(parser_id.name());
+                }
+            }
+        }
```

Comment:
```suggestion
        if_chain! {
            if language_from_syntect.raw.is_empty()
                || language_from_syntect.raw.to_lowercase() == "plain text";
            if let Some(extension) = filepath.extension();
            if let Some(parser_id) = ParserId::from_file_extension(extension.to_str().unwrap());
            then {
                return TreeSitterLanguageName::new(parser_id.name());
            }
        }
```

After adding `use if_chain::if_chain` (you can see other examples in the code), since this feature isn't stabilized yet. https://github.com/rust-lang/rust/issues/53667